### ISSUE TITLE
BADDATA

### ISSUE THREAD

**Issue by cranky-coder** [souvik1997/obdython#3]
I have a bluetooth dongle that I have hooked up to my raspberry pi.  But no matter what sensor I query I get BADDATA back.  I have tested with the torque app and it picked up the data.  

Any ideas on troubleshooting?

**Comment by souvik1997** [comment]
Hey brunkj. You could try connecting to the obd-ii adapter over the serial port and try to enter commands manually. What I have here works with my specific adapter, and I have not tested it with any other adapters.

**Comment by cranky-coder** [comment]
any recommendations on some resources to do that? :)

**Comment by SaintAnalog** [comment]
I receive the same BADDATA error. I am using the ELM327 based bluetooth OBDII adapter from Total Car Diagnostics. I was hoping since it is also based on the ELM firmware, the code would be agnostic between vendors.
I know the adapter is paired; the POWER, OBD, and PC LEDs are active on the adapter while the code runs.
Perhaps a formatting issue with the data? It looks like it is time for some debugging before I go buy the specific adapter you used in your project (iSaddle).

**Comment by souvik1997** [comment]
Yeah it's probably a formatting issue. This code is kind of a hack and I didn't really pay much attention to compatibility with other adapters.

I would use GNU screen to connect to the serial port run the commands manually to see what happens.

**Comment by SaintAnalog** [comment]
This is a hobby project of mine, so it's weekend work. I also purchased the iSaddle mini Bluetooth adapter you used in your project in hopes that it's "plug-n-play." 
I will also work on the adapter from Total Car Diagnostics and debug the data.

**Comment by SaintAnalog** [comment]
As a quick plug-n-play option, I received the iSaddle mini Bluetooth adapter that was used in your Picard project. Unfortunately, it also states "BADDATA".
I don't have the Torque app to check, but I imagine that as long as the devices are based on legit ELM firmware and chip, there should be no issue with multiple types of adapters. Total Car Diagnostics actually states they use the exact same firmware used by ELM.
Unfortunately, it will be a while for me to debug. If anyone else would like to pick up in parallel, that would be fantastic.
To summarize a bluetooth adapter from iSaddle and Total Car Diagnostics both give "BADDATA" when reading; although, both are found and connect.
